sdk/core/azure-core/src/main/java/com/azure/android/core/util/CoreUtils.java

```diff
@@ -0,0 +1,177 @@
+     * @return True if the character sequence is null or empty, false otherwise.
+     */
+    public static boolean isNullOrEmpty(CharSequence charSequence) {
+        return charSequence == null || charSequence.length() == 0;
+    }
+
+    /*/**
+     * Turns an array into a string mapping each element to a string and delimits them using a coma.
+     * @param array Array being formatted to a string.
+     * @param mapper Function that maps each element to a string.
+     * @param <T> Generic representing the type of the array.
+     * @return Array with each element mapped and delimited, otherwise null if the array is empty or null.
+     * /
+    public static <T> String arrayToString(T[] array, Function<T, String> mapper) {
+        if (isNullOrEmpty(array)) {
+            return null;
+        }
+
+        return Arrays.stream(array).map(mapper).collect(Collectors.joining(COMMA));
+    }*/
+
+    /**
+     * Returns the first instance of the given class from an array of Objects.
+     * @param args Array of objects to search through to find the first instance of the given `clazz` type.
+     * @param clazz The type trying to be found.
+     * @param <T> Generic type
+     * @return The first object of the desired type, otherwise null.
+     */
+    public static <T> T findFirstOfType(Object[] args, Class<T> clazz) {
+        if (isNullOrEmpty(args)) {
+            return null;
+        }
+
+        for (Object arg : args) {
+            if (clazz.isInstance(arg)) {
+                return clazz.cast(arg);
+            }
+        }
+
+        return null;
+    }
+
+    /*/**
+     * Extracts and combines the generic items from all the pages linked together.
+     * @param page The paged response from server holding generic items.
+     * @param context Metadata that is passed into the function that fetches the items from the next page.
+     * @param content The function which fetches items from the next page.
+     * @param <T> The type of the item being returned in the paged response.
+     * @return The publisher holding all the generic items combined.
+     * /
+    public static <T> Publisher<T> extractAndFetch(PagedResponse<T> page, Context context,
+                                                   BiFunction<String, Context, Publisher<T>> content) {
+        String nextPageLink = page.getContinuationToken();
+        if (nextPageLink == null) {
+            return Flux.fromIterable(page.getItems());
+        }
+        return Flux.fromIterable(page.getItems()).concatWith(content.apply(nextPageLink, context));
+    }*/
+
+
+    /*/**
+     * Helper method that returns an immutable {@link Map} of properties defined in {@code propertiesFileName}.
+     *
+     * @param propertiesFileName The file name defining the properties.
+     * @return an immutable {@link Map}.
+     * /
+    public static Map<String, String> getProperties(String propertiesFileName) {
```

Comment:
Left this method out since it makes use of the Java 8 Stream API and it's not used by any of the interceptors in this PR. Will address this at a later stage.